Sources/Skylab/SkylabClient.swift

```diff
@@ -87,6 +87,14 @@ public class DefaultSkylabClient : SkylabClient {
             let session = URLSession.shared
 
             let userContext = self.addContext(user:self.user)
+            
+            let userId = userContext["user_id"]
+            let deviceId = userContext["device_id"]
+            if userId == nil && deviceId == nil {
+                print("[Skylab] WARN: user id and device id are nil; no variants fetched")
+                return
```

Comment:
The early return seems fine for now as we only support amplitude_id bucketing. Without user_id and device_id, no bucketing can happen. 
In the future, we will allow bucketing on non amplitude_id as well (e.g. org_id, group_id). When we start supporting it, we would still like to warn if userId and deviceId are both missing because we won't be able to send this data automatically to Amplitude. But, skylab will continue to return the right variant in that case.